subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -412,6 +412,94 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec implements Buil
         }
     }
 
+    def "execution fails when a nested property throws an exception"() {
+        buildFile << """
+            class TaskWithFailingNestedInput extends DefaultTask {
+                @Nested
+                Object getNested() {
+                    throw new RuntimeException("BOOM")
+                }
+                
+                @Input
+                String input = "Hello"
+                
+                @OutputFile
+                File outputFile
+                
+                @TaskAction
+                void doStuff() {
+                    outputFile.text = input
+                }
+            }            
+            
+            task myTask(type: TaskWithFailingNestedInput) {
+                outputFile = file('build/output.txt')
+            }
+        """
+
+        expect:
+        fails "myTask"
+        failure.assertHasDescription("Execution failed for task ':myTask'.")
+        failure.assertHasCause("BOOM")
+    }
+
+    def "null on nested bean is validated"() {
+        buildFile << """
+            class TaskWithAbsentNestedInput extends DefaultTask {
+                @Nested
+                Object nested
+                
+                @Input
+                String input = "Hello"
+                
+                @OutputFile
+                File outputFile
+                
+                @TaskAction
+                void doStuff() {
+                    outputFile.text = input
+                }
+            }            
+            
+            task myTask(type: TaskWithAbsentNestedInput) {
+                outputFile = file('build/output.txt')
+            }            
+        """
+
+        expect:
+        fails "myTask"
+        failure.assertHasDescription("A problem was found with the configuration of task ':myTask'.")
+        failure.assertHasCause("No value has been specified for property 'nested'.")
+    }
+
+    def "null on optional nested bean is allowed"() {
+        buildFile << """
+            class TaskWithAbsentNestedInput extends DefaultTask {
+                @Nested
+                @Optional
+                Object nested
+                
+                @Input
+                String input = "Hello"
+                
+                @OutputFile
+                File outputFile
+                
+                @TaskAction
+                void doStuff() {
+                    outputFile.text = input
+                }
+            }            
+            
+            task myTask(type: TaskWithAbsentNestedInput) {
+                outputFile = file('build/output.txt')
+            }            
+        """
+
+        expect:
+        succeeds "myTask"
+    }
+
```

Comment:
Good to add these cases!

Do we have a test for infinite recursive nesting? Would be nice to see how that works with the new approach.

Reply:
I added a test. It is currently ignored, since we run into an infinite loop.